pmm-app/src/pmm-qan/panel/components/Filters/hooks/useFilters.ts

```diff
@@ -19,6 +24,23 @@ export const useFilters = (): [any, boolean, boolean] => {
       try {
         const result = await FiltersService.getQueryOverviewFiltersList(labels, from, to, columns[0]);
 
+        const filteredResult = Object.keys(result)
+          .filter((key) => !FILTERS_GROUPS.some((group) => group.dataKey === key))
+          .filter((key) => !HIDDEN_FILTER_LABELS.includes(key))
+          .reduce((obj, key) => {
+            obj[key] = result[key];
+
+            return obj;
+          }, {});
+
+        Object.keys(filteredResult).forEach((commentKey) => FILTERS_GROUPS.push({
+          name: commentKey
+            .replace(/^\w/, (c) => c.toUpperCase())
+            .replace(/_/g, ' ')
+            .substring(0, COMMENT_NAME_LENGTH),
+          dataKey: commentKey,
+        }));
```

Comment:
Again just a suggestion, but I would merge those two:
```suggestion
        const filtersGroups = Object.keys(result)
          .filter((key) => !FILTERS_GROUPS.some((group) => group.dataKey === key))
          .filter((key) => !HIDDEN_FILTER_LABELS.includes(key))
          .reduce((groups, key) => {
            groups.push({
              name: key
               .replace(/^\w/, (c) => c.toUpperCase())
               .replace(/_/g, ' ')
               .substring(0, COMMENT_NAME_LENGTH),
             dataKey: key,
            });

            return groups;
          }, [... FILTERS_GROUPS]);
```

by that you will omit pushing anything directly to the `const FILTERS_GROUPS` as well as less iteration on the result keys.
and then you can just return `filtersGroups`

Reply:
Sorry for misleading naming, but the idea was to add the comments (as result of the filtering the response by FILTERS_GROUP and HIDDEN_FILTERS_LABELS) to the existing FILTERS_GROUP.
Fixed with changing the naming.